OpenRA.Mods.Common/Traits/Passenger.cs

```diff
@@ -183,5 +183,14 @@ public void Unreserve(Actor self)
 			ReservedCargo.UnreserveSpace(self);
 			ReservedCargo = null;
 		}
+
+		void INotifyKilled.Killed(Actor self, AttackInfo e)
+		{
+			if (Transport == null || Transport.IsDead)
+				return;
+
+			Cargo c = Transport.Trait<Cargo>();
+			c.Unload(self);
```

Comment:
This is calling `Cargo.Unload` with the passenger as `self`, but that method expects `self` to be the transport!